openstack/obs/client_base.go

```diff
@@ -0,0 +1,46 @@
+package obs
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// ObsClient defines OBS client.
+type ObsClient struct {
+	conf       *config
+	httpClient *http.Client
+}
+
+// New creates a new ObsClient instance.
+func New(ak, sk, endpoint string, configurers ...Configurer) (*ObsClient, error) {
+	conf := &config{securityProvider: &securityProvider{ak: ak, sk: sk}, endpoint: endpoint}
+	conf.maxRetryCount = -1
+	conf.maxRedirectCount = -1
+	for _, configurer := range configurers {
+		configurer(conf)
+	}
+
+	if err := conf.initConfigWithDefault(); err != nil {
+		return nil, err
+	}
+	err := conf.getTransport()
+	if err != nil {
+		return nil, err
+	}
+
+	if isWarnLogEnabled() {
+		info := make([]string, 3)
+		info[0] = fmt.Sprintf("[OBS SDK Version=%s", obs_sdk_version)
+		info[1] = fmt.Sprintf("Endpoint=%s", conf.endpoint)
+		accessMode := "Virtual Hosting"
+		if conf.pathStyle {
+			accessMode = "Path"
+		}
+		info[2] = fmt.Sprintf("Access Mode=%s]", accessMode)
+		doLog(LEVEL_WARN, strings.Join(info, "];["))
+	}
+	doLog(LEVEL_DEBUG, "Create obsclient with config:\n%s\n", conf)
```

Comment:
doLog using only for obs, can you change it to some standard method?

Reply:
https://github.com/opentelekomcloud/gophertelekomcloud/blob/665dacb4afbcc8fde38ae6edb50e4558014e7f8f/openstack/obs/log.go#L266

doLog contains Log formatting methods customized for OBS.
Do you have any better ideas?